cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -253,7 +255,181 @@ EXEC strQry // SQL Injection vulnerability!
 
 DO: Practise Least Privilege - Connect to the database using an account with a minimum set of permissions required to do it's job i.e. not the sa account
 
-## A2 Weak Account management
+### OS Injection
+
+Information about OS Injection can be found on this [cheat sheet](OS_Command_Injection_Defense_Cheat_Sheet.md#net).
+
+DO: Use [System.Diagnostics.Process.Start](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.start?view=netframework-4.7.2) to call underlying OS functions.
+
+e.g
+
+``` csharp
+System.Diagnostics.Process process = new System.Diagnostics.Process();
+System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+startInfo.FileName = "validatedCommand";
+startInfo.Arguments = "validatedArg1 validatedArg2 validatedArg3";
+process.StartInfo = startInfo;
+process.Start();
+```
+
+DO: Use whitelist validation on all user supplied input. Input validation prevents improperly formed data from entering an information system. For more information please see the [Input Validation Cheat Sheet](Input_Validation_Cheat_Sheet.md).
+
+e.g Validating user input using [IPAddress.Parse Method](https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.parse?view=netframework-4.8)
+
+``` csharp
+//User input
+string ipaddress = "127.0.0.1";
+ 
+//check to make sure an ip address was provided    
+if (string.IsNullOrEmpty(address))  
+{   
+	try
+	{
+		// Create an instance of IPAddress for the specified address string (in 
+		// dotted-quad, or colon-hexadecimal notation).
+		IPAddress address = IPAddress.Parse(ipAddress);
+
+		// Display the address in standard notation.
+		return address.ToString();
+	}
+	catch(FormatException e)
+	{
+		//ipaddress is not of type IPaddress
+		...
+	}
+    ...
+}
+ ```
+ 
+### LDAP injection
+
+Almost any characters can be used in Distinguished Names. However, some must be escaped with the backslash "\" escape character. A table showing which characters that should be escaped for Active Directory can be found at the [Cheat Sheet](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/LDAP_Injection_Prevention_Cheat_Sheet.md#defense-option-1-escape-all-variables-using-the-right-ldap-encoding-function).
```

Comment:
The backslash is not rendered in the sentence `some must be escaped with the backslash "\"`, put the character between `xxx` to render it:

![PRT001](https://user-images.githubusercontent.com/1573775/61310213-2409ed80-a7f4-11e9-933a-9d9d81b5771c.png)

Reply:
There also a duplicate in the link to the LDAP cheat sheet.